cloud/ec2_ami_copy.py

```diff
@@ -0,0 +1,224 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: ec2_ami_copy
+short_description: copies AMI between AWS regions, return new image id
+description:
+    - Copies AMI from a source region to a destination region. This module has a dependency on python-boto >= 2.5
+version_added: "1.7"
+options:
+  source_region:
+    description:
+      - the source region that AMI should be copied from
+    required: true
+    default: null
+    aliases: []
+  dest_region:
+    description:
+      - the destination region that AMI should be copied to
+    required: true
+    default: null
+    aliases: []
+  source_image_id:
+    description:
+      - the id of the image in source region that should be copied
+    required: true
+    default: null
+    aliases: []
+  name:
+    description:
+      - The name of the new image to copy
+    required: false
+    default: null
+    aliases: []
+  description:
+    description:
+      - An optional human-readable string describing the contents and purpose of the new AMI.
+    required: false
+    default: null
+    aliases: []
+  wait:
+    description:
+      - wait for the copied AMI to be in state 'available' before returning.
+    required: false
+    default: "no"
+    choices: [ "yes", "no" ]
+    aliases: []
+  wait_timeout:
+    description:
+      - how long before wait gives up, in seconds
+    required: false
+    default: 1200
+    aliases: []
+  tags:
+    description:
+      - a hash/dictionary of tags to add to the new copied AMI; '{"key":"value"}' and '{"key":"value","key":"value"}'
+    required: false
+    default: null
+    aliases: []
+
+author: Amir Moulavi <amir.moulavi@gmail.com>
+extends_documentation_fragment: aws
+'''
+
+EXAMPLES = '''
+# Basic AMI Copy
+- local_action:
+    module: ec2_ami_copy
+    aws_access_key: "-----REMOVED-------"
+    aws_secret_key: "-----REMOVED-------"
+    source_region: eu-west-1
```

Comment:
I'd simply remove `aws_access_key` and `aws_secret_key` from the examples. They're not included in examples of other AWS modules, and they just clutter up the example.